generators/plugin/templates/composer.json

```diff
@@ -0,0 +1,18 @@
+{
+  "name": "<%= plugin.name.fileName %>",
+  "type": "application",
```

Comment:
To conform with general unwritten rules for WordPress plugins, I believe the type of this package should be `wordpress-plugin` instead of `application`.